app/models/update_existing_draft_edition.rb

```diff
@@ -79,4 +95,20 @@ def attributes
   def edition_attributes_from_payload
     payload.slice(*Edition::TOP_LEVEL_FIELDS)
   end
+
+  def publisher_first_published_at
```

Comment:
Oh, of course, I see what you mean... sorry, I completely missed that distinction.

I wonder if we can have some sort of mapping with something like (untested code):

```ruby
DATE_FIELDS = {
  publisher_major_updated_at: :major_updated_at,
  publisher_minor_updated_at: :minor_updated_at,
  ...
}

attributes = DATE_FIELDS.each_with_object({}) do |hsh, (internal_key, external_key)|
  hsh[internal_key] = payload[external_key] if payload[external_key].present?
end

edition.update_attributes!(attributes)
```